lib/src/model.dart

```diff
@@ -2770,13 +2787,14 @@ Map<PackageWarning, List<String>> packageWarningText = {
 
 // Something that package warnings can be called on.
 abstract class Warnable implements Locatable {
-  void warn(PackageWarning warning, [String message]);
+  void warn(PackageWarning warning, {String message, Locatable referredFrom});
 }
 
 // Something that can be located for warning purposes.
```

Comment:
This is existing code, but this comment should probably be a doc comment (`///`).